src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -42,34 +43,74 @@ public override async Task<int> ExecuteAsync()
                 {
-                    Console.WriteLine($"Could not find a build with id '{_options.Id}'");
+                    Console.WriteLine($"Could not find a build with id '{_options.Id}'.");
                     return Constants.ErrorCode;
                 }
 
-                Channel targetChannel = await UxHelpers.ResolveSingleChannel(remote, _options.Channel);
-                if (targetChannel == null)
+                if (string.IsNullOrEmpty(_options.Channel) && !_options.AddToDefaultChannels)
                 {
+                    Console.WriteLine("You need to use --channel or --default-channels to inform the channel(s) that the build should be promoted to.");
                     return Constants.ErrorCode;
                 }
 
-                if (build.Channels.Any(c => c.Id == targetChannel.Id))
+                List<Channel> targetChannels = new List<Channel>();
+
+                if (!string.IsNullOrEmpty(_options.Channel))
+                {
+                    Channel targetChannel = await UxHelpers.ResolveSingleChannel(remote, _options.Channel);
+                    if (targetChannel == null)
+                    {
+                        return Constants.ErrorCode;
+                    }
+
+                    targetChannels.Add(targetChannel);
+                }
+
+                if (_options.AddToDefaultChannels)
                 {
-                    Console.WriteLine($"Build '{build.Id}' has already been assigned to '{targetChannel.Name}'");
+                    IEnumerable<DefaultChannel> defaultChannels = await remote.GetDefaultChannelsAsync(
+                        build.GitHubRepository ?? build.AzureDevOpsRepository, 
+                        build.GitHubBranch ?? build.AzureDevOpsBranch);
+
+                    targetChannels.AddRange(
+                        defaultChannels.
+                            Where(dc => dc.Enabled).
+                            Select(dc => dc.Channel));
+                }
+
+                IEnumerable<Channel> currentChannels = build.Channels.Where(ch => targetChannels.Any(tc => tc.Id == ch.Id));
+                if (currentChannels.Any())
+                {
+                    Console.WriteLine($"The build '{build.Id}' is already on the following channel(s):");
+
+                    foreach (var channel in currentChannels)
+                    {
+                        Console.WriteLine($"\t{channel.Name}");
+                    }
+                }
+
+                if (!targetChannels.Any())
+                {
+                    Console.WriteLine($"Build '{build.Id}' is already on all target channel(s).");
                     return Constants.SuccessCode;
                 }
 
-                Console.WriteLine($"Assigning the following build to channel '{targetChannel.Name}':");
+                Console.WriteLine($"Assigning build '{build.Id}' to the following channel(s):");
+                foreach (var channel in targetChannels)
```

Comment:
I think you want to assign only channels that are not in currentChannels?

Reply:
Thanks. Good catch!